clang/lib/AST/ASTContext.cpp

```diff
@@ -10510,6 +10512,8 @@ QualType ASTContext::mergeFunctionTypes(QualType lhs, QualType rhs,
     if (lproto->getMethodQuals() != rproto->getMethodQuals())
       return {};
 
+    // TODO: (nonblocking) Does anything need to be done with FunctionEffects?
+
```

Comment:
I managed to refresh my memory of when this can be called -- it's in C, for redeclarations with attribute mismatches, and in situations like trying to build a type for the result of a conditional expression. There's a good example with how this method handles `noreturn`, which I'll copy.